### ISSUE TITLE
[BUG] adding support for app-provided monochrome/adaptive icons

### ISSUE THREAD

**Issue by RustoMCSpit** [Deletescape-Media/Lawnchair#3568]
### Describe the bug

if you set your icons to themed and switch to lawnicons and then switch back to system icons theyll stay as lawnicons, you cant get back regular monochrome system icons 

### Steps to reproduce

1) turn on themed icons
2) swap to lawnicons
3) swap back to system icons

### Expected behavior

see regular system monochrome icons

### Device information

pixel 6a,

**Comment by KTibow** [comment]
pretty sure if you want to turn off themed icons, you need to turn off themed icons, not just change the icon pack

**Comment by RustoMCSpit** [comment]
> pretty sure if you want to turn off themed icons, you need to turn off themed icons, not just change the icon pack

regardless of what pack i choose, the themed icons are the same; they are ONLY the lawnchair icon themes. i cant see 'systemicon' themed icons.

**Comment by RustoMCSpit** [comment]
https://github.com/thetwom/toc2/releases

download this app and turn on themed system icons, if it doesnt go monochrome the bug is confirmed as real because in all other launchers it does. for me, it isnt themed despite having a systemicon theming

**Comment by KTibow** [comment]
I don't know what "system monochrome" you're talking about - AFAIK you need Lawnicons to use themed icons.

**Comment by RustoMCSpit** [comment]
> I don't know what "system monochrome" you're talking about - AFAIK you need Lawnicons to use themed icons.

you dont need lawnicons to use themed icons, turn lawnchair off for a second and go to the regular launcher and turn on themed icons

**Comment by KTibow** [comment]
You're assuming everyone has themed icons built into their system launcher. Most people don't.

**Comment by RustoMCSpit** [comment]
> You're assuming everyone has themed icons built into their system launcher. Most people don't.

right, well find a launcher that lets https://github.com/thetwom/toc2/releases turn into its monochrome form and youll understand what im talkimg about

**Comment by RustoMCSpit** [comment]
if you want a screenshot i can give yu one

**Comment by KTibow** [comment]
Sure, give a screenshot of the problem.
It seems odd given that the icon pack you use for one launcher is affecting Lawnchair.

**Comment by RustoMCSpit** [comment]
> Sure, give a screenshot of the problem. It seems odd given that the icon pack you use for one launcher is affecting Lawnchair.

![_20230811_230716](https://github.com/LawnchairLauncher/lawnchair/assets/134429563/a5399ca0-44ee-4cf7-ae91-3f10f2f6c205)

**Comment by RustoMCSpit** [comment]
thats what the monochrome theme of the app looks like normally in any other launcher with system icons, lawnchair launcher is unable to do that. lawnchair launchers monochrome mode only supports its own icpn pack

**Comment by KTibow** [comment]
Can you please edit your issue title and contents to clarify that this is about adding support for app-provided monochrome/adaptive icons?

**Comment by KTibow** [comment]
still relevant

**Comment by crystalarcus** [comment]
please add the support for app provided monochrome icon, I am a developer and I can't see my apps monochrome icon in lawnchair, every time I want to change it and test how it looks in phone, i have to change home screen default app to OneUI. It would be better if Lawnchair can use monochrome provided by apps rather than 3rd party premade icons.

**Comment by KTibow** [comment]
nopedy dopedy

**Comment by abdallahmehiz** [comment]
Is this feature feasible? If not, why? Lack of launcher privileges? Lack of system APIs? Or something different?
